From 2417be12b85dedafe1464758bb18c2efd340e1a9 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Thu, 2 Jul 2015 09:45:38 -0700
Subject: [PATCH] Platform/architecture can now the overridden

The npm_config items 'platform' and 'arch' override
process.process and process.arch if present.
---
 lib/install.js                           | 5 ++++-
 lib/utils/error-handler.js               | 4 ++--
 node_modules/npm-install-checks/index.js | 8 ++++----
 3 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/lib/install.js b/lib/install.js
index 4ab248a..dfa24f4 100644
--- a/lib/install.js
+++ b/lib/install.js
@@ -1034,9 +1034,12 @@ function installOne_ (target, where, context, cb_) {
       , strict = npm.config.get("engine-strict")
       , c = npmInstallChecks
 
+    var platform = npm.config.get("platform") || process.platform;
+    var arch = npm.config.get("arch") || process.arch;
+
     chain(
         [ [c.checkEngine, target, npm.version, nodeVersion, force, strict]
-        , [c.checkPlatform, target, force]
+        , [c.checkPlatform, target, platform, arch, force]
         , [c.checkCycle, target, context.ancestors]
         , [c.checkGit, targetFolder]
         , [write, target, targetFolder, context] ]
diff --git a/lib/utils/error-handler.js b/lib/utils/error-handler.js
index 1c80ab5..7758ca6 100644
--- a/lib/utils/error-handler.js
+++ b/lib/utils/error-handler.js
@@ -259,8 +259,8 @@ function errorHandler (er) {
               ,"Not compatible with your operating system or architecture: "+er.pkgid
               ,"Valid OS:    "+er.os.join(",")
               ,"Valid Arch:  "+er.cpu.join(",")
-              ,"Actual OS:   "+process.platform
-              ,"Actual Arch: "+process.arch
+              ,"Actual OS:   "+er.platform
+              ,"Actual Arch: "+er.arch
               ].join("\n"))
     break
 
diff --git a/node_modules/npm-install-checks/index.js b/node_modules/npm-install-checks/index.js
index 10f214f..6a410e0 100644
--- a/node_modules/npm-install-checks/index.js
+++ b/node_modules/npm-install-checks/index.js
@@ -27,10 +27,8 @@ function checkEngine (target, npmVer, nodeVer, force, strict, cb) {
 }
 
 exports.checkPlatform = checkPlatform
-function checkPlatform (target, force, cb) {
-  var platform = process.platform
-    , arch = process.arch
-    , osOk = true
+function checkPlatform (target, platform, arch, force, cb) {
+  var osOk = true
     , cpuOk = true
 
   if (force) {
@@ -48,6 +46,8 @@ function checkPlatform (target, force, cb) {
     er.code = "EBADPLATFORM"
     er.os = target.os || ['any']
     er.cpu = target.cpu || ['any']
+    er.platform = platform
+    er.arch = arch
     er.pkgid = target._id
     return cb(er)
   }
-- 
2.0.1

